### ISSUE TITLE
Print image

### ISSUE THREAD

**Issue by mb720** [eXeC64/imv#249]
Hi!

I started working on an alias in imv that lets me print the current image.

The way [Zathura](https://git.pwmt.org/pwmt/zathura) handles this for PDFs is quite convenient using its `:print` command which is defined [here](https://git.pwmt.org/pwmt/zathura/-/blob/master/zathura/print.c).

My progress so far has been to convert the image with ImageMagick to a PostScript file with

    convert current_file.jpg eps2:current_file.eps

Now I'd like have a way to choose the printer. Zathura brings up the [GTK print dialog](https://developer.gnome.org/gtk3/stable/GtkPrintUnixDialog.html) for that, which would work for me.

Is there a way to do this with an alias, or does some code of imv have to change to print the current image?

**Comment by eXeC64** [comment]
I've not looked into what print helpers are available, but my expectation would be along the lines of `:exec print-command $imv_current_file`, where `print-command` is some program that launches a print dialog for printing the given file.

For the imv part of this, you could either type that manually, define a bind, or define an alias to `:print`.

It's definitely doable, assuming such a helper program exists. Print functionality will not be added to imv itself however.

**Comment by mb720** [comment]
I used [`yad`](https://www.systutorials.com/docs/linux/man/1-yad/) as that helper program and added an alias to imv's config:

    [aliases]
    print = exec yad --print --type=IMAGE --filename="$imv_current_file"

This prints the image but I noticed that the quality of the printed page is significantly worse than when printing a PDF, with zathura for example.

I converted the file into a PDF with `convert` and tried to print it with `yad --print --type=RAW --filename=test.pdf` but in the CUPS interface I got an error:

>"Filter failed"

I added a bug report [here](https://sourceforge.net/p/yad-dialog/tickets/320/).

I don't expect you to fix this issue for me, I'm just documenting my progress.
